File: CodingPractice-master/leetcode/algorithms/python/0104_MaximumDepthOfBinaryTree.py
# -*- coding: utf-8 -*-
# @Author: Yan_Daojiang
# @Date:   2019-05-14 23:00:09
# @Last Modified by:   Yan_Daojiang
# @Last Modified time: 2019-05-14 23:00:28

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def maxDepth(self, root: TreeNode) :
        if root==None:
            return 0
        else:
            return 1+max(self.maxDepth(root.left),self.maxDepth(root.right))
    
    def max(a,b)->int:
        if a>=b:
            return a
        else: 
            return b
        

File: CodingPractice-master/leetcode/algorithms/cpp/0104_MaximumDepthOfBinaryTree.cpp
/*
* @Author: Yan_Daojiang
* @Date:   2019-05-14 22:58:49
* @Last Modified by:   Yan_Daojiang
* @Last Modified time: 2019-05-14 22:59:22
*/

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* root) {
        if (root==NULL)
            return 0;
        else 
            return 1+max(maxDepth(root->left),maxDepth(root->right));
    }
public:
    int max(int num1,int num2){
        if(num1>=num2)
            return num1;
        else 
            return num2;
    }
};